sharedbuffers/mapped_struct.py

```diff
@@ -1708,6 +1712,65 @@ def _c_search_hkey_ui64(hkey, pindex, stride0, length, hint):
 
     @cython.cfunc
     @cython.locals(
+        hkey = cython.longlong,
+        mkey = cython.longlong,
+        lo = cython.size_t, hi = cython.size_t, length = cython.size_t,
+        mid = cython.size_t, mid2 = cython.size_t, stride0 = cython.size_t, hint = cython.size_t,
+        pindex = cython.p_char, skip = cython.size_t)
+    @cython.returns(cython.size_t)
+    def _c_search_hkey_i64(hkey, pindex, stride0, length, hint):
+        hi = length
+        lo = 0
+        if lo < hi:
+            # First iteration a quick guess assuming uniform distribution of keys
+            mid = min(hint, hi-1)
```

Comment:
What I'm saying is: this is a private method, used inside the class. you're doing defensive programming on a private method. If it were public you have a point. 
The calls to this method use hint as lo+hi//2 so the only way to get a segmentation fault is if lo and hi are 0, but you have a lo<hi check.

As you said: hinted_bsearch takes the externally supplied value and in that case, and since that method is the public one you should check the min there.